developers/blobstream-rollups.md

```diff
@@ -0,0 +1,123 @@
+Then, users/rollup full nodes will be able to query that data, and in case of a dispute, they can create a fraud proof.
+
+The difference between the above construction and this one is that the proofs used will be different:
+when using a sequence of spans,
+the inclusion proofs will be straight from the rollup data up to the [data root tuple root](https://github.com/celestiaorg/blobstream-contracts/blob/master/README.md#how-it-works).
+However, in the case of using the `height` and the `share commitment`,
+an extra step would need to be done when posting the header to the settlement contract.
+This step is proving that the provided commitment is part of the Celestia block (referenced by its height).
+Then, fraud proofs will need to prove the following:
+
+- share inclusion to the share commitment: meaning creating two merkle proofs:
+  - share merkle proof up to the [subtree root](https://celestiaorg.github.io/celestia-app/specs/data_square_layout.html#blob-share-commitment-rules)
+    corresponding to that share
+  - subtree root merkle proof to the [share commitment](https://celestiaorg.github.io/celestia-app/specs/data_square_layout.html#blob-share-commitment-rules)
+- share commitment inclusion to the data root tuple root: meaning four merkle proofs:
+  - subtree roots merkle proofs to the share commitment: to make sure the subtree roots are valid
+  - subtree roots merkle proofs up to the row roots: to prove that the subtree roots belong to a set of rows
+  - row roots proofs to the data root
+  - data root tuple proof to the data root tuple
+
+Once these are valid, the rollup contract can proceed to parse the share and verify the contested claim.
+
+Note: **Generating/verifying share commitment proofs is still not supported.
+It still needs tooling to generate the proofs on the node side,
+and verifying them on the Solidity side which will be built in the upcoming months.
+Thus, the [sequence of spans](#optimistic-rollups-that-uses-a-sequence-of-spans) approach is preferable at the moment.
+Also, the gas cost of verifying all the above proofs needs to be investigated.**
+
+#### Protobuf deserialization contract
+
+Another alternative when using the share commitment to reference the rollup data is to parse the PFB,
+i.e., the protobuf serialized PayForBlob transaction,
+inside the rollup settlement contract and verify that the parsed share commitment is the same.
+Then, proving the inclusion of the PFB to the data root tuple root.
+More details on this can be found in [ADR-011](https://github.com/celestiaorg/celestia-app/blob/main/docs/architecture/adr-011-optimistic-blob-size-independent-inclusion-proofs-and-pfb-fraud-proofs.md). 
+
+The issue with this approach is needing a library to parse the protobuf in solidity and the gas cost that comes with it.
+Otherwise, proving the PFB to the data root tuple root is cheaper than proving the blob,
+because it needs only proving two shares.
```

Comment:
this separate section is good! I think we should try to do this with all of the ways to prove data. Agnostic to how rollups actually use them. Most rollup devs know how their proving systems work, so if we just tell them about the different ways to prove data, then they can connect that to their proving system.

we can ofc discuss our thoughts on how these things work in a zk or optimistic rollup later as well